src/App.tsx

```diff
@@ -22,30 +22,61 @@ const Header = (
 );
 
 const App: React.FC = () => {
-  const [victimList, setVictimList] = useState([]);
+  const [victimList, setVictimList] = useState();
   const [isDataReady, setIsDataReady] = useState(true);
-  const [isResponseOk, setIsResponseOk] = useState(true);
+  const [isResponseOk, setIsResponseOk] = useState();
 
   const fetchDetails = async (data: SearchData) => {
     setIsDataReady(false);
     console.log(
       "sending request to: ",
       process.env.REACT_APP_BACKEND_URL + `/find/victim/byName/${data.name}`
     );
-    const response = await fetch(
+    fetch(
       process.env.REACT_APP_BACKEND_URL + `/find/victim/byName/${data.name}`
-    );
+    )
+    .then((response) => {
+      if (response.ok) {
+        // Returning the JSON response if HTTP status response code is 200/201
+        const contentType = response.headers.get("content-type");
+        if (contentType && contentType.indexOf("application/json") !== -1) {
+          // Returns JSON object if the response is of type JSON.
+          return response.json();
+        } else {
+          // Returns string if the response is of type string.
+          return response.text();
+        }
```

Comment:
Not all the API return JSON value until it's specified in backend. We should check for Emergency response service error message 
which is currently not returning any response as such and that response need not be a JSON value also. Since, we aren't getting any such response from the API. I have written that code for future refactoring.

I.e, Proceed only if I receive a JSON object as response else consider it as an error message.